Data/Primitive/ByteArray.hs

```diff
@@ -413,6 +457,23 @@ copyByteArrayToPtr (Ptr dst#) (ByteArray src#) soff sz
   where
   siz# = sizeOf# (undefined :: a)
 
+-- | Copy from an unmanaged pointer address to a byte array. These must not
+-- overlap. The offset and length are given in elements, not in bytes.
+--
+-- /Note:/ this function does not do bounds or overlap checking.
+copyPtrToByteArray
+  :: forall m a. (PrimMonad m, PrimState m ~ RealWorld, Prim a)
+  => MutableByteArray RealWorld -- ^ destination array
+  -> Int -- ^ offset into destination array, interpreted as elements of type @a@
+  -> Ptr a -- ^ source
+  -> Int -- ^ number of elements to copy
+  -> m ()
+{-# INLINE copyPtrToByteArray #-}
+copyPtrToByteArray (MutableByteArray dst#) soff (Ptr src#) sz
+  = primitive_ (copyAddrToByteArray# src# dst# (unI# soff *# siz# ) (unI# sz))
+  where
+  siz# = sizeOf# (undefined :: a)
+
```

Comment:
Surely we want to export `copyPtrToByteArray`, whether with this argument order or a different one, but that's for another PR.

Reply:
There is https://hackage.haskell.org/package/primitive-0.7.3.0/docs/Data-Primitive-Ptr.html#v:copyPtrToMutablePrimArray

Is it the same?

Reply:
Aha! I guess we just need to move its definition.

Reply:
:+1:

Reply:
Inm the name of consistency, should copyPtrToMutablePrimArray be exported from `PrimArray` module as well?

Reply:
Yes, I think so.

Reply:
Added a commit to export `copyPtrToMutablePrimArray` from `PrimArray`.